reactor-netty5-http/src/main/java/reactor/netty5/http/Cookies.java

```diff
@@ -86,35 +77,24 @@ public Map<CharSequence, Set<Cookie>> getCachedCookies() {
 			}
 		}
 
-		List<String> allCookieHeaders = allCookieHeaders();
-		Map<String, Set<Cookie>> cookies = new HashMap<>();
-		for (String aCookieHeader : allCookieHeaders) {
-			Set<Cookie> decode;
-			if (isClientChannel) {
-				final Cookie c = ((ClientCookieDecoder) decoder).decode(aCookieHeader);
-				if (c == null) {
-					continue;
-				}
-				Set<Cookie> existingCookiesOfName = cookies.computeIfAbsent(c.name(), k -> new HashSet<>());
-				existingCookiesOfName.add(c);
+		Map<java.lang.CharSequence, Set<HttpCookiePair>> cookies = new HashMap<>();
+		Iterator<HttpSetCookie> setCookieItr = nettyHeaders.getSetCookiesIterator();
+		while (setCookieItr.hasNext()) {
+			try {
+				HttpSetCookie setCookie = setCookieItr.next();
+				Set<HttpCookiePair> existingCookiesOfName = (Set<HttpCookiePair>) cookies.computeIfAbsent(setCookie.name(), k -> new HashSet<>());
+				existingCookiesOfName.add(setCookie);
 			}
-			else {
-				decode = ((ServerCookieDecoder) decoder).decode(aCookieHeader);
-				for (Cookie cookie : decode) {
-					Set<Cookie> existingCookiesOfName = cookies.computeIfAbsent(cookie.name(), k -> new HashSet<>());
-					existingCookiesOfName.add(cookie);
-				}
+			catch (HeaderValidationException err) {
+				// ignore invalid Set-Cookie header
```

Comment:
Added a DEBUG trace. However, it's not possible to log the exact Set-Cookie headers which could not be decoded. To work around, I'm logging the index number of the invalid Set-Cookie header.
Also, we need to catch the IllegalArgumentException instead of the HeaderValidationException, because I have seen that sometimes, an IllegalArgumentException may be thrown.